user.rb

```diff
@@ -0,0 +1,13 @@
+class User
+  attr_reader :slackname, :firstname, :lastname, :about_me, :track, :course, :timezone
+
+  def initialize(slackname, firstname, lastname, about_me, track, course, timezone)
+    @slackname = slackname
+    @firstname = firstname
+    @lastname = lastname
+    @about_me = about_me
+    @track = track
```

Comment:
I've defined the track, course and timezone instance variables as text. They could just as easily be their own classes ("entities" ), but I thought that might be overkill for now.

Reply:
I need further explain on this comment...

Reply:
```
  def tuple_to_user(tuple)
    slackname = tuple["slackname"]
    firstname = tuple["firstname"]
    lastname = tuple["lastname"]
    about_me = tuple["about_me"]
    track_code = tuple["track"]
    track_name = ...
    course = tuple["course"]
    timezone = tuple["timezone"]

    User.new(
      slackname,
      firstname,
      lastname,
      about_me,
      Track.new(track_code, track_name),
      Course.new(...),
      Timezone.new(...)
    )
  end
```